src/chapters/az-cli/walkthrough_az-cli.rst

```diff
@@ -0,0 +1,537 @@
+
+- ``-n``: <name>-linux-vm
+- ``--size``: Standard_B2s
+- ``--admin-username``: student
+- ``--image``: the image URN [stored in a variable]
+
+.. note::
+
+    It is important that you use these exact values so that it is easier to help you if something goes wrong along the way.
+
+Let's create our VM! Note that this command will take some time to complete.
+
+.. sourcecode:: powershell
+    :caption: Windows/PowerShell
+
+    > az vm create -n <name>-linux-vm --size "Standard_B2s" --image "$ImageURN" --admin-username "student" --assign-identity
+
+.. sourcecode:: bash
+    :caption: Linux/BASH
+
+    $ az vm create -n <name>-linux-vm --size "Standard_B2s" --image "$image_urn" --admin-username "student" --assign-identity
+
+.. admonition:: note
+
+  If you receive the following error output:
+
+  .. sourcecode:: bash
+
+    An RSA key file or key value must be supplied to SSH Key Value.
+    
+    You can use --generate-ssh-keys to let CLI generate one for you
+
+  You can fix this by reissuing the command and appending ``--generate-ssh-keys`` after ``--assign-identity``. We will learn about SSH, RSA keys and how they relate to this message in later lessons.    
+
+You should receive an output like this:
+
+.. sourcecode:: bash
+
+    {
+        "fqdns": "",
+        "id": "/subscriptions/<subscription ID>/resourceGroups/<name>-cli-wt-rg/providers/Microsoft.Compute/virtualMachines/<name>-linux-vm",
+        "identity": {
+            "systemAssignedIdentity": "<vm object ID>",
+            "userAssignedIdentities": {}
+        },
+        "location": "eastus",
+        "macAddress": "00-0D-3A-18-98-5F",
+        "powerState": "VM running",
+        "privateIpAddress": "10.0.0.4",
+        "publicIpAddress": "13.72.111.180",
+        "resourceGroup": "<name>-cli-wt-rg",
+        "zones": ""
+    }
+
+Notice how the default resource group value you set earlier was automatically included along with the subscription and location.  
+
+Configuring
+-----------
+
+Before we continue let's set this VM as the default:
+
+.. sourcecode:: bash
+
+    $ az configure --default vm=<name>-linux-vm
+
+Next let's use the VM ``show`` Command to view all of the details of our new VM. The ``show`` Command requires the following Arguments:
+
+- ``-n``: VM name (``--ids`` can be used in place of the name)
+- ``-g``: the resource group the VM is in
+- ``--subscription``: the subscription the VM is a part of
+
+Since we have configured default values for each of these arguments we do not need to provide any of them to issue the command:
+
+.. sourcecode:: bash
+
+    $ az vm show
+
+If you configured the default VM correctly you should receive a lengthy output object representing the state and configuration of the new VM. We will make use of the ``show`` Command when granting access to the KeyVault in the following section.
+
+KeyVault Secrets
+================
+
+As our final step we will provision and configure our KeyVault.
```

Comment:
Maybe add a one or two sentences reminder of what KeyVault is, and why we need to set it up.